modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/rest/api/customSchema/SCIM2CustomSchemaMeTestCase.java

```diff
@@ -147,17 +152,33 @@ private void createClaims() throws Exception {
         loginLogoutClient = new LoginLogoutClient(context);
         cookie = loginLogoutClient.login();
         claimMetadataManagementServiceClient = new ClaimMetadataManagementServiceClient(backendURL, cookie);
-        int claimDialect = claimMetadataManagementServiceClient.getClaimDialects().length;
+        ClaimDialectDTO[] claimDialects = claimMetadataManagementServiceClient.getClaimDialects();
 
-        // Set custom schema dialect.
-        ClaimDialectDTO claimDialectDTO = new ClaimDialectDTO();
-        claimDialectDTO.setClaimDialectURI(CUSTOM_SCHEMA_URI);
-        claimMetadataManagementServiceClient.addClaimDialect(claimDialectDTO);
+        boolean isCustomSchemaDialectExist = Arrays.stream(claimDialects)
+                .anyMatch(claimDialect -> StringUtils.equals(claimDialect.getClaimDialectURI(), CUSTOM_SCHEMA_URI));
+
+        if (!isCustomSchemaDialectExist) {
```

Comment:
If the claim dialect is existing, the purpose of this testcase will not be fulfilled... What is the possibility of this being already available? I believe, we need to avoid this check.

Reply:
With this PR (https://github.com/wso2/carbon-identity-framework/pull/5661), we register the custom claim dialect and add external claims. In the test case it tries to create the same claim dialect again, which leads to the issue.

Reply:
Previously, the custom claim dialect was only created in IS when a custom SCIM attribute was added. With the above PR, some SCIM attributes are now packed by default under the custom SCIM claim dialect. As a result, the test case is trying to recreate an already existing dialect, which is causing the issue. I think now we can verify whether the custom test attributes are correctly created within the existing custom claim dialect.

Reply:
Either we check the custom dialect addition or check the already existing dialect. We cannot do both here.

Reply:
Rather than checking custom dialect addition or existence, focus here is moved to test if following custom external claims are properly created within the custom claim dialect, as this test method not only create the claim dialect but also add following external claims to the custom claim dialect.
 - urn:scim:wso2:schema:country
 - urn:scim:wso2:schema:manager
 - urn:scim:wso2:schema:manager.email
 
```
Assert.assertTrue(Arrays.stream(externalClaimDTOs)
                .anyMatch(claim -> StringUtils.equals(claim.getExternalClaimURI(), COUNTRY_CLAIM_ATTRIBUTE_URI)));
                
Assert.assertTrue(Arrays.stream(externalClaimDTOs)
        .anyMatch(claim -> StringUtils.equals(claim.getExternalClaimURI(), MANAGER_CLAIM_ATTRIBUTE_URI)));
        
Assert.assertTrue(Arrays.stream(externalClaimDTOs)
                .anyMatch(claim -> StringUtils.equals(claim.getExternalClaimURI(), MANAGER_EMAIL_CLAIM_ATTRIBUTE_URI)));
```

Reply:
"urn:scim:wso2:schema" claim dialect adding part is removed, and test description is updated now properly.